Fall back to copying file on Windows

sivadeilra opened the pull request:
Creating a symbolic link requires administrator privileges (by default)
on Windows. This commit falls back to copying a file if creating a
symbolic link fails, on Windows.

marco-c commented on src/producer.rs:
```diff
@@ -285,11 +285,14 @@ impl Archive {
+                os::windows::fs::symlink_file(&src_path, &path)
+                    .or_else(|_| std::fs::copy(&src_path, &path).map(|_| ()))
+                    .unwrap_or_else(|_| {
+                        panic!("Failed to create a symlink {:?} -> {:?}", src_path, path)
```
Could you update the error message?

sivadeilra replied:
Sure. What would you like it to be?  Something like `"Failed to create symlink / copy file"` or `"Failed to copy file"`, or something else?

marco-c replied:
If we implement the symlink creation warning I suggested below, we can say "Failed to copy file".
If we don't implement the warning, we can say "Failed to create symlink or copy file".

calixteman replied:
It's a bit hard... with big projects it can become awful.
I wonder if should use an option to allow that...
If some people are using grcov with a kind of big project and they aren't aware about that stuff then each time it'll run a forest will burn. And if we've an option (`--copy-when-no-symlink` or whatever) then the first run will fail because the option isn't used, and then with a good error message they can decide to copy or to allow symlink creations in configuring windows correctly.

marco-c commented:
Given that this is going to result in slowdown, maybe we should warn the user (only once) when we are not able to create symlinks and suggest them to run grcov with admin privileges.

sivadeilra commented:
I like the idea of printing a warning. I've updated the PR. It will now print a warning, but only once per grcov. I added a section to `README.md` explaining how to enable symbolic links, and the warning should point people to that text.

marco-c commented on src/symlink.rs:
```diff
@@ -0,0 +1,34 @@
+) -> std::io::Result<()> {
+    use std::sync::atomic::{AtomicBool, Ordering::SeqCst};
+
+    static HAVE_PRINTED_WARNING: AtomicBool = AtomicBool::new(false);
```
An AtomicBool shouldn't be necessary because the producer is single threaded, but better to be safe than sorry in case we change that in the future.

calixteman reviewed (approved):
LGTM.
Thanks for doing this: it's valuable (especially because we are linux devs).
